src/Functions/FunctionsHashing.h

```diff
@@ -231,12 +231,23 @@ struct SipHash64Impl
 struct SipHash128Impl
 {
     static constexpr auto name = "sipHash128";
-    enum { length = 16 };
 
-    static void apply(const char * begin, const size_t size, unsigned char * out_char_data)
+    using ReturnType = UInt128;
+
+    static UInt128 combineHashes(UInt128 h1, UInt128 h2)
+    {
+        UInt128 hashes[] = {h1, h2};
+        return apply(reinterpret_cast<const char *>(hashes), 32);
+    }
+
+    static UInt128 apply(const char * data, const size_t size)
     {
-        sipHash128(begin, size, reinterpret_cast<char*>(out_char_data));
+        char bytes[16];
+        sipHash128(data, size, bytes);
```

Comment:
Let's make get rid of `bytes` variable. Change `sipHash128` return type to `UInt128`.